include/timer.h

```diff
@@ -93,37 +93,37 @@ static constexpr int precise_delay_duration_us = 100;
 static inline void DelayPrecise(const int milliseconds)
 {
 	// The estimate of how long the sleep should take (microseconds)
-	static double estimate = 5e-5;
+	static auto estimate = 5e-5f;
 	// Use the estimate value as the default mean time taken
-	static double mean = 5e-5;
-	static double m2 = 0;
+	static auto mean = 5e-5f;
+	static auto m2 = 0.0f;
 	static int64_t count = 1;
 
 	// Original code operated on seconds, convert
-	double seconds = milliseconds / 1e3;
+	auto seconds = static_cast<float>(milliseconds) / 1e3f;
 
 	// sleep as long as we can, then spinlock the rest
 	while (seconds > estimate) {
 		const auto start = GetTicksUs();
 		DelayUs(precise_delay_duration_us);
 		// Original code operated on seconds, convert
-		const double observed = GetTicksUsSince(start) / 1e6;
+		const auto observed = static_cast<float>(GetTicksUsSince(start)) / 1e6f;
 		seconds -= observed;
 
 		++count;
-		const double delta = observed - mean;
-		mean += delta / count;
+		const auto delta = observed - mean;
+		mean += delta / static_cast<float>(count);
 		m2 += delta * (observed - mean);
-		const double stddev = std::sqrt(m2 / (count - 1));
+		const auto stddev = std::sqrt(m2 / static_cast<float>(count - 1));
 		estimate = mean + stddev;
 	}
 
     // spin lock
     const auto spin_start = GetTicksUs();
-	const int spin_remain = static_cast<int>(seconds * 1e6);
-	do {
-		std::this_thread::yield();
-	} while (GetTicksUsSince(spin_start) <= spin_remain);
+    const int spin_remain = static_cast<int>(seconds * 1e6f);
+    do {
+	    std::this_thread::yield();
+    } while (GetTicksUsSince(spin_start) <= spin_remain);
```

Comment:
Use tabs instead of spaces, please.

Reply:
Thanks for catching this @dreamer. 

This is very strange; the only formatting I applied was via clang-autoformat (using the usual script). 

I wonder if some of the original formatting also used spaces, and auto-format is yielding to that? (And not outright replacing spaces with tabs?). 

Will go through the the PRs formatting fully tomorrow and fix it.

Reply:
clang-format is indeed ingecting spaces into previously tabbed `case:` blocks:

![2021-07-29_09-57](https://user-images.githubusercontent.com/1557255/127533636-b832d429-55ab-45d1-90bc-a1ebc68abe40.png)

I've had it perform fine in previous case statements; so will try to figure out what's throwing it off.

Reply:
They probably changed it because spaces are indeed correct.

https://insanelab.com/blog/notes/spaces-vs-tabs/

<DUCKING>

Reply:
In the above, you see can the lower `case:` sections became indented further, as if they were children of the prior case. 

I moved the top-level switch brace down (left aligned with the switch), and then clang got the entire switch correct:

![2021-07-29_10-02](https://user-images.githubusercontent.com/1557255/127534240-e16aac4e-8150-4745-81d4-01c079ed2a66.png)